install.py

```diff
@@ -1020,6 +1020,7 @@ def driver():
         dest="thread_count",
         nargs="?",
         type=int,
+        default=os.environ.get("THREAD_COUNT"),
```

Comment:
Could we use conda build's built-in [`CPU_COUNT`](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-set-during-the-build-process) envvar instead of defining our own?

Reply:
Replied to the same question in the cunumeric PR.